### ISSUE TITLE
Bug: [member-ordering] update docs to remove invalid config

### ISSUE THREAD

**Issue by mike-lischke** [typescript-eslint/typescript-eslint#6136]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=KYDwDg9gTgLgBAbzgXwFCoMYQHYGd4CWcAvHAAwDcqQA&eslintrc=N4KABGBEAOCGBOBnApvSAuKABALgT2mUQGN4BLaHAWiIBsyA7HAejiVUgBpwo2V4A8pTIB7BogxhQECDHgiAVsmI5JkHCTEAzMgHMAdAsRiuPWcYCu8YsgAqBZGoC2IgCYXaj7jKhkn0emIyHABlHHIVSXCLZG8ZSGUnWAAxZFgcKyJJaR8oIwAPKPgYszAAXzjZRNgANVREUQZJACYABmbmngqeGFoLXUYJTABtUshcBxJyShpEeiZIHgBdOMhizyGpMYnCKYpqOkYWJ2QnACNUKhF4V1RGXUlR3KhUeTRKmRznyFutWA9VCNSt8GroGOlMqZnj4YBYzoEqIgcOkyMQqDpkLRXFDobJoPIcMpCa5EcicKj0WRMdiPt98WQAG7pZCklFojFYnG42Hwim3YjXZkkjk04G5OQiQkqZAk-mC4mU6lc6FyRnMqhy+BCxWc2ni6BwhGDZEMGw60W4vEEokyqjG2Cmlki5V08hMwl28Qms3OvUwg28tGwM5IrUqc0u-XW6Uk4Oh2Dh31i-1u9Vx8IJ6hJy28Q0U7OWiVShUF7n091Oqm65PxJFsiN++L2x0NmuydNhrNVi2FzXa0sqgd0vNogVe4oqa6RlOSm0ksehiyT95t3ipj0L6LL6fxTcTnBTxt4kes8lok44AAWbh3VtnMdPFIv157ZfXLLrZ6oz5vR9zgY1ZR5VtH9XxVfF7wVPsFVA281zVD1oJA5Ar1-VceSNL0HTNWC-yLOdPTrFtcPQ8t1WbHCUJfOCMIpDtM2-Ki0JzfCH3o8MSJYsiPXY6hYJrJYxTKUpBIgYSyiAA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkeAhpgObp4VTokBMADF15AA04GpHiwAJgFcM7SJyLoAHm2E0o9VKlgB3AFJlKeTFPRrRAKxJK5mdPWiqRtK0oBiDvLEwBPOQAshMGcoVzdMeiZ4dCI8D0dvP0pIcMjo2KEQyFgpPAARZEw5ADoAehy8RFzSzPVIO3EAN3QAYQR0vEMwY1NzWhIczGh0AFl6RHYesyzOEckZdABJWKxYCaMTabqAI3oSdABVTFQSmsFgusR6PH8u4Av1KDtPdgBtELqyokl0AH1xaQYEilRDPRylOBIa6QD5gAC6fVEYLwAFoJAg3rDaF8fv95kCQciIQgrmxYXCQgBfRGQEjGZCOSabGkAM28wzapGQdJieBae2QRCYyzcaHQADl6NEulMaQCFgAlTiwVC5QhEOTfCToWpI5WoZoGYhzQE6ja9LLclXXdASE0LGXMrLfACi8GQbHNW1E30WSFQDI9AEEiEluk66iQpIhQZwSH7EAHoMHQ8ttUoXZhMN5HRbI4NhgrYLBPVASENdbQA9tMY8ONw+AJEbR0fAYeoEVkSABrZCIAAyyG2LX86Gg3aZedEykQWGQHU0eTH3muOfYLM0+xpmm0OgAyj5YqOCNAlxuZHgE95OpPvf16Y4AArZ2eYPA+ZYe5CaZAAL2u6rrpud4cO6+TLhEXiYCMrD0BI1z0EBqBbs6sAeFoNzZlITD+Py+zxkQe46B60CBJQG7IegIjUiIkCCtAqrarWojltApQAFTsRx7bwuY8hKAxUhMZQrwUpSQA

### Repro Code

```typescript
The issue is not related to any code and the above playground link does not show the error.
```

### ESLint Config

```javascript
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "tsconfig.json",
    "sourceType": "module",
    "implicitStrict": true,
    "ecmaFeatures": {
      "jsx": true
    },
    "ecmaVersion": 2022
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/member-ordering": [
      "error",
      {
        "default": [
          "signature",
          "public-static-field",
          "protected-static-field",
          "private-static-field",
          "public-decorated-field",
          "protected-decorated-field",
          "private-decorated-field",
          "public-instance-field",
          "protected-instance-field",
          "private-instance-field",
          "public-abstract-field",
          "protected-abstract-field",
          "private-abstract-field",
          "public-field",
          "protected-field",
          "private-field",
          "static-field",
          "instance-field",
          "abstract-field",
          "decorated-field",
          "field",
          "public-constructor",
          "protected-constructor",
          "private-constructor",
          "constructor",
          "public-static-method",
          "protected-static-method",
          "private-static-method",
          "public-decorated-method",
          "protected-decorated-method",
          "private-decorated-method",
          "public-instance-method",
          "protected-instance-method",
          "private-instance-method",
          "public-abstract-method",
          "protected-abstract-method",
          "private-abstract-method"
        ]
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "target": "es2022",
    "module": "esnext",
    "allowJs": true,
    "jsx": "react",
    "jsxFactory": "h", 
    "jsxFragmentFactory": "Fragment",
    "outDir": "./output/",
    "removeComments": true,
    "sourceMap": true,
    "esModuleInterop": true,
    "baseUrl": "./", 
    "paths": { 
      "react": [
        "./node_modules/preact/compat"
      ],
      "react-dom": [
        "./node_modules/preact/compat"
      ]
    },
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "noImplicitAny": true,
    "suppressImplicitAnyIndexErrors": true,
    "sourceRoot": "src",
    "lib": [
      "es2022",
      "dom"
    ],
    "skipLibCheck": true,
    "experimentalDecorators": false,
    "allowSyntheticDefaultImports": true,
    "strictPropertyInitialization": false,
    "emitDecoratorMetadata": false,
    "noFallthroughCasesInSwitch": false
  },
  "include": [
    "src/**/*"
  ],
  "exclude": []
}
```

### Expected Result

I expected no error from eslint, as the project is currently empty.

### Actual Result

```
 eslint src


Oops! Something went wrong! :(

ESLint: 8.28.0

Error: .eslintrc.json:
        Configuration for rule "@typescript-eslint/member-ordering" is invalid:
        Value ["signature","public-static-field","protected-static-field","private-static-field","public-decorated-field","protected-decorated-field","private-decorated-field","public-instance-field","protected-instance-field","private-instance-field","public-abstract-field","protected-abstract-field","public-field","protected-field","private-field","static-field","instance-field","abstract-field","decorated-field","field","public-constructor","protected-constructor","private-constructor","constructor","public-static-method","protected-static-method","private-static-method","public-decorated-method","protected-decorated-method","private-decorated-method","public-instance-method","protected-instance-method","private-instance-method","public-abstract-method","protected-abstract-method","private-abstract-method"] should be string.
        Value ["signature","public-static-field","protected-static-field","private-static-field","public-decorated-field","protected-decorated-field","private-decorated-field","public-instance-field","protected-instance-field","private-instance-field","public-abstract-field","protected-abstract-field","public-field","protected-field","private-field","static-field","instance-field","abstract-field","decorated-field","field","public-constructor","protected-constructor","private-constructor","constructor","public-static-method","protected-static-method","private-static-method","public-decorated-method","protected-decorated-method","private-decorated-method","public-instance-method","protected-instance-method","private-instance-method","public-abstract-method","protected-abstract-method","private-abstract-method"] should be equal to one of the allowed values.
        Value "private-abstract-method" should be equal to one of the allowed values.
        Value "private-abstract-method" should be array.
        Value "private-abstract-method" should match exactly one schema in oneOf.
        Value ["signature","public-static-field","protected-static-field","private-static-field","public-decorated-field","protected-decorated-field","private-decorated-field","public-instance-field","protected-instance-field","private-instance-field","public-abstract-field","protected-abstract-field","public-field","protected-field","private-field","static-field","instance-field","abstract-field","decorated-field","field","public-constructor","protected-constructor","private-constructor","constructor","public-static-method","protected-static-method","private-static-method","public-decorated-method","protected-decorated-method","private-decorated-method","public-instance-method","protected-instance-method","private-instance-method","public-abstract-method","protected-abstract-method","private-abstract-method"] should be object.
        Value ["signature","public-static-field","protected-static-field","private-static-field","public-decorated-field","protected-decorated-field","private-decorated-field","public-instance-field","protected-instance-field","private-instance-field","public-abstract-field","protected-abstract-field","public-field","protected-field","private-field","static-field","instance-field","abstract-field","decorated-field","field","public-constructor","protected-constructor","private-constructor","constructor","public-static-method","protected-static-method","private-static-method","public-decorated-method","protected-decorated-method","private-decorated-method","public-instance-method","protected-instance-method","private-instance-method","public-abstract-method","protected-abstract-method","private-abstract-method"] should match exactly one schema in oneOf.

    at ConfigValidator.validateRuleOptions (/Users/mike/Downloads/Test/preact/preact2/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2027:23)
    at /Users/mike/Downloads/Test/preact/preact2/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2082:18
    at Array.forEach (<anonymous>)
    at ConfigValidator.validateRules (/Users/mike/Downloads/Test/preact/preact2/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2079:34)
    at ConfigValidator.validateConfigArray (/Users/mike/Downloads/Test/preact/preact2/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2205:18)
    at CascadingConfigArrayFactory._finalizeConfigArray (/Users/mike/Downloads/Test/preact/preact2/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3962:23)
    at CascadingConfigArrayFactory.getConfigArrayForFile (/Users/mike/Downloads/Test/preact/preact2/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3768:21)
    at FileEnumerator._iterateFilesRecursive (/Users/mike/Downloads/Test/preact/preact2/node_modules/eslint/lib/cli-engine/file-enumerator.js:448:49)
    at _iterateFilesRecursive.next (<anonymous>)
    at FileEnumerator.iterateFiles (/Users/mike/Downloads/Test/preact/preact2/node_modules/eslint/lib/cli-engine/file-enumerator.js:299:49)

```

### Additional Info

What's interesting here is that 2 specific entries cause the problem: "private-abstract-field" and "private-abstract-method". If I comment them out everything goes fine.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.45.0` |
| `@typescript-eslint/parser`            | `5.45.0` |
| `TypeScript`                           | `4.9.3` |
| `ESLint`                               | `8.28.0` |
| `node`                                 | `16.18.1` |

**Comment by Josh-Cena** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/6132

**Comment by Josh-Cena** [comment]
Private abstract properties don't exist, so these are invalid config entries to begin with. We simply fixed the bad behavior.

**Comment by mike-lischke** [comment]
OK, fine, but don't forget to fix the [documentation](https://typescript-eslint.io/rules/member-ordering/#default-configuration), which mentions these entries (which I thoughtless took over, when I added member ordering) :-)

**Comment by jdforsythe** [comment]
We were hit with the same issue here. I get they're invalid - but we just copied the default and modified it to our liking. The PR which removed these didn't seem to mention anything about why they were removed, either.

**Comment by SaumyaBhushan** [comment]
@bradzacher @mike-lischke Should I comment this out  "private-abstract-field" and "private-abstract-method" from the [documentation](https://typescript-eslint.io/rules/member-ordering/#default-configuration)

**Comment by bradzacher** [comment]
@SaumyaBhushan yup - we should remove all mentions of `private-abstract-*` from the docs entirely!

**Comment by sviat9440** [comment]
I've already done that, haven't I?
Maybe just update the documentation?

https://github.com/typescript-eslint/typescript-eslint/pull/5859/files#diff-76f7820640084772a4df6af9a91b71d64e8e2ec5f1933a727b52317660f47858

**Comment by sviat9440** [comment]
![image](https://user-images.githubusercontent.com/13107421/204978634-33b55e5b-8e72-42e1-a10f-5cf2e852f43f.png)
Now there is an old version of the TSEslint and documentation

**Comment by SaumyaBhushan** [comment]
Hi, @sviat9440 I got confused. It looks like you have already made those changes, so where do I need to do them?

**Comment by bradzacher** [comment]
cc @JoshuaKGoldberg / @JamesHenry - looks like the website docs are pinned to an old version? Likely related to the changes we made to prevent unreleased changes from showing up.

**Comment by JamesHenry** [comment]
Sorry folks, we had some incorrect netlify config, which has now been resolved so this won't happen again, the website is now inline with the latest release, `5.45.0`
